packages/backend/src/modules/tracked-txs/repositories/TrackedTxsConfigsRepository.ts

```diff
@@ -0,0 +1,170 @@
+import { Logger } from '@l2beat/backend-tools'
+import {
+  notUndefined,
+  ProjectId,
+  TrackedTxsConfigSubtype,
+  TrackedTxsConfigType,
+  UnixTime,
+} from '@l2beat/shared-pure'
+import { Knex } from 'knex'
+import { TrackedTxsConfigRow } from 'knex/types/tables'
+
+import {
+  BaseRepository,
+  CheckConvention,
+} from '../../../peripherals/database/BaseRepository'
+import { Database } from '../../../peripherals/database/Database'
+import { TrackedTxId } from '../types/TrackedTxId'
+import {
+  TrackedTxConfigEntry,
+  TrackedTxUseWithId,
+} from '../types/TrackedTxsConfig'
+
+export interface TrackedTxsConfigRecord {
+  id: TrackedTxId
+  projectId: ProjectId
+  type: TrackedTxsConfigType
+  subtype?: TrackedTxsConfigSubtype
+  sinceTimestamp: UnixTime
+  untilTimestamp?: UnixTime
+  lastSyncedTimestamp?: UnixTime
+  debugInfo: string
+}
+
+export class TrackedTxsConfigsRepository extends BaseRepository {
+  constructor(database: Database, logger: Logger) {
+    super(database, logger)
+    this.autoWrap<CheckConvention<TrackedTxsConfigsRepository>>(this)
+  }
+
+  async getAll(): Promise<TrackedTxsConfigRecord[]> {
+    const knex = await this.knex()
+    const rows = await knex('tracked_txs_configs')
+    return rows.map(toRecord)
+  }
+
+  async addMany(
+    records: TrackedTxsConfigRecord[],
+    trx?: Knex.Transaction,
+  ): Promise<TrackedTxId[]> {
+    const knex = await this.knex(trx)
+
+    const insertedRows = await knex('tracked_txs_configs')
+      .insert(records.map(toRow))
+      .returning('id')
+
+    return insertedRows
+      .map((row) => (row.id ? TrackedTxId.unsafe(row.id) : undefined))
+      .filter(notUndefined)
+  }
+
+  // TODO: (tracked_txs_status) function useful for implementing tracked txs status
```

Comment:
we can try leaving it there, but I feel it would be better to describe it in the task in Linear, with maybe hash to the code version that had this code in liveness repository